AEPAssurance/Source/AssuranceStateManager.swift

```diff
@@ -0,0 +1,215 @@
+/*
+ Copyright 2022 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPCore
+import AEPServices
+import Foundation
+import UIKit
+
+class AssuranceStateManager {
+
+    let datastore = NamedCollectionDataStore(name: AssuranceConstants.EXTENSION_NAME)
+
+    /// sessionID represents a unique identifier of a Project Griffon session to which this library attempts to connect.
+    /// This property gets set when a deep-link initiating an assurance session is received.
+    /// The presence of valid sessionID variable in `AssuranceStateManager` denotes a connection to Assurance session has been initiated.
+    /// Note: The presence of sessionID does not denote active socket connection
+    /// The absence of sessionID denotes a session connection is terminated or cancelled.
+    var sessionId: String? {
+        get {
+            datastore.getString(key: AssuranceConstants.DataStoreKeys.SESSION_ID)
+        }
+        set {
+            datastore.set(key: AssuranceConstants.DataStoreKeys.SESSION_ID, value: newValue)
+        }
+    }
+
+    private let DEFAULT_ENVIRONMENT = AssuranceEnvironment.prod
+
+    /// environment property defines the current environment to which the assurance session is connecting.
+    /// This value should not be confused with launch environment.
+    /// This value denotes the environment of Griffon services and UI running to which connection is made.
+    /// This value is obtained from as a query parameter of deeplink URL.
+    var environment: AssuranceEnvironment {
+        get {
+            AssuranceEnvironment.init(envString: datastore.getString(key: AssuranceConstants.DataStoreKeys.ENVIRONMENT) ?? DEFAULT_ENVIRONMENT.rawValue)
```

Comment:
I think its better to keep `environment` as a computed variable (from the datastore) and not have a backing variable that stores the value 

And also getter to the environment variable is used only while initiating a connection to griffon. Hence minimum performance impact

Reply:
After session management part of the refactor is implemented (1:1 session : socket connection), `AssuranceEnvironment` can just be fetched once from the saved connection url and injected  into the AssuranceSession (as a local variable) to be used during the connection. This will likely remove the need to hold it in the AssuranceStateManager, isn't it?

Reply:
that should work. Will keep updated along with AssuranceSessionManagement changes